### PR TITLE
[TECH] Profile page

### PR DESCRIPTION
**Pull request by sadoffToxir**
Created new reducer and actions, also did some updates

### CODE REVIEW

File: src/assets/images/defaultImages/defaultAvatar.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...434 C195.487565,56.9663626 191.276535,45.9419513 183.679824,38.9488198 Z" id="react-path-230"></path><filter x="-0.8%" y="-2.0%" width="101.5%" height="108.0%" filterUnits="objectBoundingBox" id="react-filter-227"><feOffset dx="0" dy="2" in="SourceAlpha" result="shadowOffsetOuter1"></feOffset><feColorMatrix values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.16 0" type="matrix" in="shadowOffsetOuter1" result="shadowMatrixOuter1"></feColorMatrix><feMerge><feMergeNode in="shadowMatrixOuter1"></feMergeNode><feMergeNode in="SourceGraphic"></feMergeNode></feMerge></filter></defs><mask id="react-mask-229" fill="white"><use xlink:href="#react-path-231"></use></mask><g id="Mask"></g><g id="Top/Short-Hair/Short-Waved" mask="url(#react-mask-229)"><g transform="translate(-1.000000, 0.000000)"><mask id="react-mask-228" fill="white"><use xlink:href="#react-path-230"></use></mask><use id="Short-Hair" stroke="none" fill="#28354B" fill-rule="evenodd" xlink:href="#react-path-230"></use><g id="Skin/👶🏽-03-Brown" mask="url(#react-mask-228)" fill="#4A312C"><g transform="translate(0.000000, 0.000000) " id="Color"><rect x="0" y="0" width="264" height="280"></rect></g></g></g></g></g></g></g></g></g></svg>
```
Comment (mtalstykh-exadel) [inline src/assets/images/defaultImages/defaultAvatar.svg]: use material icons

**Comment by sadoffToxir** [comment]
Solved issues with structure and code

File: src/components/profile/profileInfo/profileInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+const ProfileInfo = () => {
+  const avatar = useSelector((state) => state.profile.avatar);
+  const fullname = useSelector((state) => state.profile.fullname);
+  const occupation = useSelector((state) => state.profile.occupation);
+  const email = useSelector((state) => state.profile.email);
+
+  return <div>
+    {avatar ? fullname : <Avatar></Avatar>}
+    <h2>{fullname ? fullname : 'Fullname'}</h2>
```
Comment (mtalstykh-exadel) [inline src/components/profile/profileInfo/profileInfo.jsx]: better to show nothing if fullname is empty

File: src/components/profile/profileInfo/profileInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { Avatar } from '@material-ui/core';
+
+const ProfileInfo = () => {
+  const avatar = useSelector((state) => state.profile.avatar);
+  const fullname = useSelector((state) => state.profile.fullname);
+  const occupation = useSelector((state) => state.profile.occupation);
+  const email = useSelector((state) => state.profile.email);
+
+  return <div>
+    {avatar ? fullname : <Avatar></Avatar>}
```
Comment (mtalstykh-exadel) [inline src/components/profile/profileInfo/profileInfo.jsx]: show fullname if we have a non empty avatar? sounds strange

Comment (mtalstykh-exadel) [reply src/components/profile/profileInfo/profileInfo.jsx]: And you can write <Avatar/> not <Avatar></Avatar>

File: src/pages/profile/profile.jsx [hunk 4]
```diff
@@ -1,59 +1,16 @@
...
-        color="primary"
-        disableElevation
-        className={classes.testButtons}
-      >
-        <Link to="/jwt">JWT</Link>
-      </Button>
-    </div>
-  );
+  return <>
+		{/* Header */}
```
Comment (mtalstykh-exadel) [inline src/pages/profile/profile.jsx]: you don't need header here. We will use <Layout /> component for it

Comment (mtalstykh-exadel) [reply src/pages/profile/profile.jsx]: get it from develop

File: src/store/actions/loginActions.js [hunk 5]
```diff
@@ -14,6 +15,9 @@ export const fetchLoginData = (dt) => (dispatch, getState) => {
   }
   dispatch(fetchLoginStart(dt));
   return getLoginData(dt)
-    .then((authData) => dispatch(fetchLoginSuccess(authData)))
+    .then((authData) => {
+      dispatch(fetchLoginSuccess(authData));
+      dispatch(setProfileData());
```
Comment (mtalstykh-exadel) [inline src/store/actions/loginActions.js]: seems you missed params for setProfileData(?, ?, ?)

File: src/store/store.js [hunk 6]
```diff
@@ -3,10 +3,12 @@ import thunk from "redux-thunk";
 import { combineReducers } from 'redux';
 import { counterReducer } from './reducers/counterReducers';
 import { loginReducer } from './reducers/loginReducers';
+import { profileReducer } from './reducers/profileReducers';
 
 const rootReducers = combineReducers({
   counter: counterReducer,
```
Comment (mtalstykh-exadel) [inline src/store/store.js]: lets remove counterReducer from our project. We don't need it anymore

**Comment by sadoffToxir** [comment]
stylized components and added test history with filter

**Comment by sadoffToxir** [comment]
Solved issues with .map

File: src/components/profile/profileInfo/profileInfo.jsx (hunk 2 above)
Comment (sadoffToxir) [reply src/components/profile/profileInfo/profileInfo.jsx]: I wrote this like a jammer

File: src/components/profile/profileInfo/profileInfo.jsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+const ProfileInfo = () => {
+  const useStyles = makeStyles((theme) => ({
+    large: {
+      width: theme.spacing(10),
+      height: theme.spacing(10),
+      margin: '5px 46px 0 0',
+      float: 'left',
+    },
+  }));
```
Comment (mtalstykh-exadel) [inline src/components/profile/profileInfo/profileInfo.jsx]: use just './profileInfo.scss';

**Comment by sadoffToxir** [comment]
corrected some code

File: src/components/profile/testsInfo/testsData/testsData.jsx [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    { id: 'level', label: 'Level', minWidth: 50, align: 'center' },
+    { id: 'assigned', label: 'Assigned', minWidth: 140, align: 'center' },
+    { id: 'deadline', label: 'Deadline', minWidth: 140, align: 'center', },
+    { id: 'dateVerified', label: 'Date verified', minWidth: 140, align: 'center', },
+    { id: 'status', label: 'Status', minWidth: 50, align: 'center', },
+    { id: 'result', label: 'Result', minWidth: 80, align: 'center', },
+    { id: 'action', label: 'Action', minWidth: 100, align: 'center', },
+  ];
+
+  const useStyles = makeStyles({
```
Comment (mtalstykh-exadel) [inline src/components/profile/testsInfo/testsData/testsData.jsx]: use scss files to style the component

File: src/components/profile/testsInfo/testsData/testsData.jsx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  const [rowsPerPage, setRowsPerPage] = React.useState(100);
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event) => {
+    setRowsPerPage(+event.target.value);
+    setPage(0);
+  };
+  let keysForComps = 1;
```
Comment (mtalstykh-exadel) [inline src/components/profile/testsInfo/testsData/testsData.jsx]: what is "Comps"?

File: src/components/profile/testsInfo/testsInfo.jsx [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { useState } from 'react';
+import TestsData from './testsData/testsData';
+import TestInfoSearchForm from './testsInfoSearchForm/testsInfoSearchForm';
+
+const TestsInfo = () => {
+  const [filter, setFilter] = useState(null);
+  return <div>
+    <div>
```
Comment (mtalstykh-exadel) [inline src/components/profile/testsInfo/testsInfo.jsx]: why so many divs without classes?

File: src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.jsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import PropTypes from 'prop-types';
+import './testsInfoSearchForm.scss';
+
+const TestInfoSearchForm = (props) => {
+  const useStyles = makeStyles((theme) => ({
+    formControl: {
+      margin: '0 10px 0 0',
+      minWidth: 100,
```
Comment (mtalstykh-exadel) [inline src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.jsx]: use testsInfoSearchForm.scss for styling

File: src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.jsx [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+          <InputLabel htmlFor="level">Level</InputLabel>
+          <Select native value={formik.values.level} name='level' onChange={formik.handleChange} label="Level"
+            inputProps={{ name: 'level' }}>
+            <option aria-label="All" value = {null} />
+            <option>A1</option>
+            <option>A2</option>
+            <option>B1</option>
+            <option>B2</option>
+            <option>C1</option>
+            <option>C2</option>
```
Comment (mtalstykh-exadel) [inline src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.jsx]: You can move levels to constant and generate list using "map" function. Ask Alesia about the approach

File: src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.jsx [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+          </Select>
+        </FormControl>
+
+        <Button color="primary" variant="contained" type="submit" style={{ height: 40 }}>
```
Comment (mtalstykh-exadel) [inline src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.jsx]: use scss files for styling

File: src/store/actions/loginActions.js [hunk 14]
```diff
@@ -14,6 +15,9 @@ export const fetchLoginData = (dt) => (dispatch, getState) => {
   }
   dispatch(fetchLoginStart(dt));
   return getLoginData(dt)
```
Comment (mtalstykh-exadel) [inline src/store/actions/loginActions.js]: is "dt" just a short version of "data"?

Comment (sadoffToxir) [reply src/store/actions/loginActions.js]: yes

**Comment by sadoffToxir** [comment]
realized comments

**Comment by sadoffToxir** [comment]
resolved changes

File: src/components/profile/profileInfo/profileInfo.jsx [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+import { Avatar } from '@material-ui/core';
+import './profileInfo.scss';
+
+const ProfileInfo = () => {
+  const avatar = useSelector((state) => state.profile.avatar);
+  const fullname = useSelector((state) => state.profile.fullname);
+  const occupation = useSelector((state) => state.profile.occupation);
+  const email = useSelector((state) => state.profile.email);
+
+  return <div className='profileInfo'>
```
Comment (tensenrey) [inline src/components/profile/profileInfo/profileInfo.jsx]: please use return (
   <code>
)

File: src/components/profile/testsInfo/testsInfo.jsx [hunk 16]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { useState } from 'react';
+import TestsData from './testsData/testsData';
+import TestInfoSearchForm from './testsInfoSearchForm/testsInfoSearchForm';
+
+const TestsInfo = () => {
+  const [filter, setFilter] = useState(null);
+  return <div>
```
Comment (tensenrey) [inline src/components/profile/testsInfo/testsInfo.jsx]: please use return (

)

File: src/pages/profile/profile.jsx [hunk 17]
```diff
@@ -1,59 +1,15 @@
...
-      </Button>
-      <Button
-        variant="contained"
-        color="primary"
-        disableElevation
-        className={classes.testButtons}
-      >
-        <Link to="/jwt">JWT</Link>
-      </Button>
+  return <>
```
Comment (tensenrey) [inline src/pages/profile/profile.jsx]: please use return (

)

**Comment by sadoffToxir** [comment]
updated changes

File: src/index.js [hunk 18]
```diff
@@ -4,14 +4,21 @@ import App from './App';
...
+import { I18nProvider } from '@lingui/react';
 import { BrowserRouter } from 'react-router-dom';
 import { Provider } from 'react-redux';
 import store from './store/store';
+import { defineLang } from './utils/lang-service.js';
 
 const AppContainer = () => {
+  defineLang();
+  
   return <BrowserRouter>
```
Comment (tensenrey) [inline src/index.js]: please use return (
)

**Comment by sadoffToxir** [comment]
corrected comments

**Comment by sadoffToxir** [comment]
updated files

**Comment by sadoffToxir** [comment]
changed size of rows

**Review by kseniyaLine — DISMISSED** [review]

File: src/components/profile/profileInfo/profileInfo.scss [hunk 19]
```diff
@@ -0,0 +1,25 @@
+.profileInfo {
+  width: 100%;
+  height: 100%;
+  float: left;
+  margin-bottom: 100px;
+}
+
+.profileData {
```
Comment (mtalstykh-exadel) [inline src/components/profile/profileInfo/profileInfo.scss]: use style nesting

File: src/components/profile/profileInfo/profileInfo.jsx [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+
+const ProfileInfo = () => {
+  const avatar = useSelector((state) => state.profile.avatar);
+  const fullname = useSelector((state) => state.profile.fullname);
+  const occupation = useSelector((state) => state.profile.occupation);
+  const email = useSelector((state) => state.profile.email);
+
+  return (
+    <div className='profileInfo'>
+      {avatar ? avatar : <Avatar className='avatar1' />}
```
Comment (mtalstykh-exadel) [inline src/components/profile/profileInfo/profileInfo.jsx]: change the className

File: src/components/profile/testsInfo/testsData/testsData.jsx [hunk 21]
```diff
@@ -0,0 +1,91 @@
...
+    { id: 'dateVerified', label: 'Date verified', minWidth: 140, align: 'center', },
+    { id: 'status', label: 'Status', minWidth: 50, align: 'center', },
+    { id: 'result', label: 'Result', minWidth: 80, align: 'center', },
+    { id: 'action', label: 'Action', minWidth: 100, align: 'center', },
+  ];
+  const filteredRows = rows.filter((r) => props.filter ? r.level === props.filter : r);
+  let keysForColumns = 1;
+
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(100);
```
Comment (mtalstykh-exadel) [inline src/components/profile/testsInfo/testsData/testsData.jsx]: use a hooks

https://learn-reactjs.ru/core/hooks/state-hook

File: src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.jsx [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+
+  const formik = useFormik({ initialValues: { level: '' }, validationSchema: null, onSubmit });
+  return (
+    <div className='searchForm'>
+      <form onSubmit={formik.handleSubmit}>
+        <FormControl variant="outlined" className='formControl' size='small'>
+          <InputLabel htmlFor="level">Level</InputLabel>
+          <Select native value={formik.values.level} name='level' onChange={formik.handleChange} label="Level"
+            inputProps={{ name: 'level' }}>
+            <option aria-label="All" value={null} />
```
Comment (mtalstykh-exadel) [inline src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.jsx]: use <MenuItem /> instead of option ( like Alesia did )

File: src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.jsx [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+            {testLevels.map((o) => <option key={testLevels.indexOf(o)}>{o}</option>)}
```
Comment (mtalstykh-exadel) [inline src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.jsx]: the "o" parametr looks bad) use at least "el" or something else

File: src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.scss [hunk 24]
```diff
@@ -0,0 +1,13 @@
+.searchForm {
+  width: 200px;
+  margin: 15px 0;
+}
+
+.submitButton {
```
Comment (mtalstykh-exadel) [inline src/components/profile/testsInfo/testsInfoSearchForm/testsInfoSearchForm.scss]: use style nesting

File: src/pages/main/main.jsx [hunk 25]
```diff
@@ -2,10 +2,12 @@ import React from 'react';
 import TestLevelsSelector from '../../components/testLevelSelector/testLevelSelector';
 import Layout from '../../components/layout/Layout';
 import './main.scss';
+import { Link } from 'react-router-dom';
 
 const Main = () => {
   return (
     <Layout>
+      <Link to='/profile'>Profile</Link>
```
Comment (mtalstykh-exadel) [inline src/pages/main/main.jsx]: why did you add it here?

File: src/pages/profile/profile.jsx [hunk 26]
```diff
@@ -1,58 +1,16 @@
...
-      >
-        <Link to="/jwt">JWT</Link>
-      </Button>
-    </div>
+    <>
+      <div className='profileRow'>
+        <ProfileInfo />
+        <TestsInfo />
+      </div>
+    </>
```
Comment (mtalstykh-exadel) [inline src/pages/profile/profile.jsx]: you dont need a react fragment here

File: src/pages/profile/profile.scss [hunk 27]
```diff
@@ -0,0 +1,9 @@
+.profileRow {
+  width: 70%;
+  margin: 10px auto;
+}
+
+.avatar1 {
```
Comment (mtalstykh-exadel) [inline src/pages/profile/profile.scss]: just an avatar

**Comment by sadoffToxir** [comment]
realised some features

**Comment by sadoffToxir** [comment]
realised autoSubmit

File: src/components/LoginForm/LoginForm.jsx [hunk 28]
```diff
@@ -0,0 +1,63 @@
...
+  const dispatch = useDispatch();
+
+  // function 'submit' where will be all actions after submitting
+  const submit = (values, { setSubmitting }) => {
+    dispatch(fetchLoginData(values));
+    setSubmitting(false);
+  };
+  return (
+    <div className="loginForm">
+      {loading && <img src={preloader} />}
```
Comment (mtalstykh-exadel) [inline src/components/LoginForm/LoginForm.jsx]: can we use a spinner inside [log in] button?

File: src/components/testsInfo/TestsData/testsData.jsx [hunk 29]
```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
```
Comment (mtalstykh-exadel) [inline src/components/testsInfo/TestsData/testsData.jsx:1]: seems you didn't rename part of path -> src/components/testsInfo/TestsData/testsData.jsx

File: src/components/testsInfo/TestsInfo.jsx [hunk 30]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
```
Comment (mtalstykh-exadel) [inline src/components/testsInfo/TestsInfo.jsx]: same here -> src/components/testsInfo/TestsInfo.jsx

File: src/components/testsInfo/TestsInfoSearchForm/TestsInfoSearchForm.jsx [hunk 31]
```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react';
+import { useFormik } from 'formik';
+import InputLabel from '@material-ui/core/InputLabel';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import PropTypes from 'prop-types';
+import './TestsInfoSearchForm.scss';
+
+const TestInfoSearchForm = (props) => {
```
Comment (mtalstykh-exadel) [inline src/components/testsInfo/TestsInfoSearchForm/TestsInfoSearchForm.jsx:9]: you can import setFilter using destructuring assignment -> const TestInfoSearchForm = ({ setFilter })

File: src/components/testsInfo/TestsInfoSearchForm/TestsInfoSearchForm.jsx [hunk 32]
```diff
@@ -0,0 +1,43 @@
...
+  const onSubmit = (values) => {
+    props.setFilter(values.level);
+  };
+
+  const formik = useFormik({ initialValues: { level: '' }, validationSchema: null, onSubmit });
+
+  useEffect(() => {
+    formik.submitForm();
+  }, [formik.values]);
+  return (
```
Comment (mtalstykh-exadel) [inline src/components/testsInfo/TestsInfoSearchForm/TestsInfoSearchForm.jsx:28]: leave an empty string before return

**Comment by sadoffToxir** [comment]
styled loginPage

**Comment by sadoffToxir** [comment]
styled loginPage

**Comment by sadoffToxir** [comment]
removed border styles in login

**Comment by sadoffToxir** [comment]
renamed styles

**Comment by sadoffToxir** [comment]
Corrected paths and made some changes in logging in

File: src/components/LoginForm/dataTunk.js [hunk 33]
```diff
@@ -0,0 +1,16 @@
...
+const rand = () => Math.random() < .9;
+
+const getLoginData = (data) => {
+  return timeout(1500).then(() =>
+    rand()
+      ? Promise.resolve(data)
+      : Promise.reject(new Error('Login Error')));
+};
+
+export default getLoginData;
```
Comment (mtalstykh-exadel) [inline src/components/LoginForm/dataTunk.js]: remove the file

File: src/pages/login/Login.jsx [hunk 34]
```diff
@@ -1,17 +1,19 @@
...
 import { Redirect } from "react-router-dom";
 
 const Login = () => {
   const auth = useSelector((state) => state.auth.isAuth);
   if (auth) return <Redirect to='/' />;
   return (
-    <div className='login'>
-      <img src={logo} alt="" className='logo' />
-      <LoginForm />
+    <div className='wrapper'>
```
Comment (mtalstykh-exadel) [inline src/pages/login/Login.jsx]: use another class for the className (at least "login-wrapper")

**Comment by sadoffToxir** [comment]
removed datathunk and renamed classes

File: src/components/LoginForm/LoginForm.jsx [hunk 35]
```diff
@@ -0,0 +1,76 @@
...
+import { CircularProgress } from "@material-ui/core";
+
+// filtering and checking what the user has entered into forms
+const validationSchema = Yup.object().shape({
+  email: Yup.string().email("Invalid email").required("Required"),
+  password: Yup.string().required("Required"),
+});
+
+const LoginForm = () => {
+  const [language, setLanguage] = useState('en');
```
Comment (fshovchko) [inline src/components/LoginForm/LoginForm.jsx]: Use switchLang service from utils/lang-service.js to switch language

**Comment by sadoffToxir** [comment]
made translation

File: src/components/LoginForm/LoginForm.jsx [hunk 36]
```diff
@@ -0,0 +1,79 @@
...
+  password: Yup.string().required("Required"),
+});
+
+const LoginForm = () => {
+  const error = useSelector((state) => state.auth.error);
+  const loading = useSelector((state) => state.auth.loading);
+
+  const dispatch = useDispatch();
+
+  // function 'submit' where will be all actions after submitting
```
Comment (tensenrey) [inline src/components/LoginForm/LoginForm.jsx]: what's the point in this comment

File: src/components/LoginForm/LoginForm.jsx [hunk 37]
```diff
@@ -0,0 +1,79 @@
...
+
+  const changeLang = (e) => {
+    e.currentTarget.outerText === 'EN' ? switchLang('ru') : switchLang('en');
+  };
+  return (
+    <div className="loginForm">
+      <Formik initialValues={{ email: "", password: "" }} validationSchema={validationSchema} onSubmit={submit} >
+        {({ values, errors, touched, handleChange, handleBlur, handleSubmit, isSubmitting, }) => (
+          <form onSubmit={handleSubmit} noValidate autoComplete="off" >
+            {/* form for email */}
```
Comment (tensenrey) [inline src/components/LoginForm/LoginForm.jsx]: what is the email form?
please write a correct comment

File: src/components/LoginForm/LoginForm.jsx [hunk 38]
```diff
@@ -0,0 +1,79 @@
...
+            <div className="fieldsWrapper">
+              <TextField id="outlined-basic" label='Email' variant="outlined" type="email" name="email"
+                onChange={handleChange} onBlur={handleBlur} value={values.email} className='textFields' />
+            </div>
+            {errors.email && touched.email ? errors.email === 'R'
+            ? <Trans>Required field</Trans>
+            : <Trans>Invalid email</Trans> : null}
+
+            {/* form for password */}
```
Comment (tensenrey) [inline src/components/LoginForm/LoginForm.jsx]: what is the password form?
please write a correct comment

File: src/locales/en/messages.po [hunk 39]
```diff
@@ -91,3 +116,7 @@ msgstr "No new notificiations for you"
 #: src/pages/main/main.jsx:13
 msgid "testLevel"
 msgstr "Choose a test according to the expected level of English. If you don't know your level, then start with the first test."
+
+#: src/components/LoginForm/LoginForm.jsx:45
+#~ msgid "{0}"
+#~ msgstr "{0}"
```
Comment (fshovchko) [inline src/locales/en/messages.po]: what is it used for?

**Comment by sadoffToxir** [comment]
removed unused comments

**Review by mtalstykh-exadel — APPROVED** [review]

**Review by tensenrey — APPROVED** [review]
